src/compiler/parser.ts

```diff
@@ -4757,16 +4757,22 @@ module ts {
             var amdDependencies: string[] = [];
             var amdModuleName: string;
 
-            // Keep scanning all the leading trivia in the file until we get to something that 
-            // isn't trivia.  Any single line comment will be analyzed to see if it is a 
-            // reference comment.
+            // Have we found anything that isn't a comment or whitespace?
+            var hasEncounteredNonTrivia = false;
+
+            // Scan the file for single line comments to analyse for references comments
             while (true) {
                 var kind = triviaScanner.scan();
+
+                if (kind === SyntaxKind.EndOfFileToken) {
+                    break;
+                }
                 if (kind === SyntaxKind.WhitespaceTrivia || kind === SyntaxKind.NewLineTrivia || kind === SyntaxKind.MultiLineCommentTrivia) {
                     continue;
                 }
                 if (kind !== SyntaxKind.SingleLineCommentTrivia) {
-                    break;

```

Comment:
Whoops, I made an incorrect assumption which was clarified by @CyrusNajmabadi:

> The batch compiler effectively ignores all comments, except for the ones before the very first token.

I thought we did a complete search of comments for references.